libraries/rush-lib/src/api/Subspace.ts

```diff
@@ -321,4 +325,71 @@ export class Subspace {
   public _addProject(project: RushConfigurationProject): void {
     this._projects.push(project);
   }
+
+  /**
+   * Returns hash value of injected dependencies in related package.json.
+   * @beta
+   */
+  public getPackageJsonInjectedDependenciesHash(): string | undefined {
+    const allPackageJson: IPackageJson[] = [];
+
+    const relatedProjects: RushConfigurationProject[] = [];
+    const subspacePnpmfileShimSettings: ISubspacePnpmfileShimSettings =
+      SubspacePnpmfileConfiguration.getSubspacePnpmfileShimSettings(this._rushConfiguration, this);
+
+    for (const rushProject of this.getProjects()) {
+      const injectedDependencies: Array<string> =
+        subspacePnpmfileShimSettings?.subspaceProjects[rushProject.packageName]?.injectedDependencies || [];
+      if (injectedDependencies.length === 0) {
+        continue;
+      }
+
+      const injectedDependencySet: Set<string> = new Set(injectedDependencies);
+
+      for (const dependencyProject of rushProject.dependencyProjects) {
+        if (injectedDependencySet.has(dependencyProject.packageName)) {
+          relatedProjects.push(dependencyProject);
+        }
+      }
+    }
+
+    // this means no injected dependencies found for current subspace
+    if (relatedProjects.length === 0) {
+      return undefined;
+    }
+
+    // get all related package.json
+    while (relatedProjects.length > 0) {
+      const rushProject: RushConfigurationProject = relatedProjects.pop()!;
+      allPackageJson.push(this._getImportantFieldsInPackageJson(rushProject.packageJson));
+      relatedProjects.push(...rushProject.dependencyProjects);
+    }
+
+    allPackageJson.sort((pa, pb) => pa.name.localeCompare(pb.name));
```

Comment:
`localeCompare` may produce different results on machines with different locale settings.

```suggestion
    const collator: Intl.Collator = new Intl.Collator('en');
    x.compare
    allPackageJson.sort(collator.compare);
```